### PR TITLE
Feature/openhelpd 100 faq review changes languages

### PR DESCRIPTION
**Pull request by jpaidoussi**
This PR fixes EPIC-OPENHELPD-15-PR and implements most of Rowan's feedback.
 
The parts of Rowan's feedback not implement are:
1. using a HSTORE field  for FAQ translations instead of the parent-child relationship.
2. using smartmin's built in CSV import

The reasons are:
1. For each FAQ we need to store three things: the translated question, the translated answer and the language code. Using an HSTORE would mean having to use the same parent question for each translation and that is not ideal.
2. Smartmin's CSV import expects the CSV file to be one row per item and for the columns to match the field names of the model. WCL's CSV file puts all the translations on the same row of the CSV file as the parent FAQ. I am deferring to Mike's knowledge of the problem space here and he said this is a good format they have shown to work well in practice.

For ease of reviewing this PR I have based it against develop rather than the ancient EPIC-OPENHELPD-15-PR branch. However once it is reviewed and everyone is happy I will rebase it onto of EPIC-OPENHELPD-15-PR and that is where it will be merged to as that is what the upstream PR is based on.

### CODE REVIEW

**Comment by jpaidoussi** [comment]
@rudigiesler @smn @KaitCrawford @miltontony Ready for review.

File: casepro/msgs/forms.py [hunk 1]
```diff
@@ -79,3 +80,45 @@ def clean_keywords(self):
...
+        else:
+            parent = None
+
+        if parent is None and labels is None:
+            raise forms.ValidationError(_("Labels are required if no Parent is selected"))
+
+        if parent is not None:
+            labels = parent.labels.all()
+
+        # TODO 19: Update Translation FAQ labels when Parent FAQ labels are updated
```
Comment (smn) [inline casepro/msgs/forms.py]: Still relevant `TODO` ?

Comment (jpaidoussi) [inline casepro/msgs/forms.py]: I think so, I didn't add any functionality to their work on this.

File: casepro/msgs/tasks.py [hunk 2]
```diff
@@ -89,3 +101,83 @@ def reply_export(export_id):
...
+        except Exception as e:
+            task.log("Label %s does not exist" % labelstring)
+            raise e
+    return list(labels)
+
+
+@task(track_started=True)
+def faq_csv_import(org, task_id):  # pragma: no cover
+    task = ImportTask.objects.get(pk=task_id)
+    log = StringIO()
```
Comment (smn) [inline casepro/msgs/tasks.py]: Nothing seems to get written to this `StringIO` instance?

File: casepro/msgs/views.py [hunk 3]
```diff
@@ -5,27 +5,47 @@
...
+    self.log("Queued import at %s" % now())
+    self.save(update_fields=['import_log'])
+
+    # trigger task
+    result = faq_csv_import.delay(org, self.pk)
+
+    self.task_id = result.task_id
+    self.save(update_fields=['task_id'])
+
+ImportTask.start = override_start
```
Comment (smn) [inline casepro/msgs/views.py]: What are we patching here instead of subclasssing?

Comment (jpaidoussi) [inline casepro/msgs/views.py]: That is all WCL's code. I just fixed the tests and refactored the language model into a field

File: casepro/msgs/tasks.py (hunk 2 above)
Comment (jpaidoussi) [inline casepro/msgs/tasks.py]: Indeed it doesn't seem like they use it anywhere. Will remove it

File: casepro/msgs/views.py (hunk 3 above)
Comment (smn) [inline casepro/msgs/views.py]: ok :)

**Comment by smn** [comment]
I'm about as happy as I can be with this PR so :+1: from me. The size of it (and how old it is) makes reviewing more challenging than it should be and so I wouldn't be surprised if more, smaller PRs, fixes will come out of this one getting merged in.

**Pull request merged**